### ISSUE TITLE
error in Recursively sort

### ISSUE THREAD

**Issue by johnmcfarlane** [pascalre/vscode-yaml-sort#48]
When I attempt to sort my yaml file, I receive a dialog box with the following message:

> "Command 'YAML Sort: Recursively sort YAML files' resulted in an error (Cannot read properties of undefined (reading 'fsPath'))"

Screenshot:
![image](https://user-images.githubusercontent.com/734007/164970223-dc9d86ed-443c-43e1-8c28-1c04dd6db3c0.png)

Running WSL2 Ubuntu on Windows 11.

```sh
uname -a
Linux Carbon 5.10.102.1-microsoft-standard-WSL2 #1 SMP Wed Mar 2 00:30:59 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

About VS Code:

> Version: 1.66.2 (user setup)
Commit: dfd34e8260c270da74b5c2d86d61aee4b6d56977
Date: 2022-04-11T07:46:01.075Z
Electron: 17.2.0
Chromium: 98.0.4758.109
Node.js: 16.13.0
V8: 9.8.177.11-electron.0
OS: Windows_NT x64 10.0.22000

YAML file:

```yaml
---
- key: cert-dcl16-c.NewSuffixes
  value: L;LL;LU;LLU
- key: cppcoreguidelines-non-private-member-variables-in-classes.IgnoreClassesWithAllMemberVariablesBeingPublic
  value: '1'
- key: google-readability-braces-around-statements.ShortStatementLines
  value: '1'
- key: google-readability-function-size.StatementThreshold
  value: '800'
- key: google-readability-namespace-comments.ShortNamespaceLines
  value: '10'
- key: google-readability-namespace-comments.SpacesBeforeComments
  value: '2'
- key: modernize-loop-convert.MaxCopySize
  value: '16'
- key: modernize-loop-convert.MinConfidence
  value: reasonable
- key: modernize-loop-convert.NamingStyle
  value: CamelCase
- key: modernize-pass-by-value.IncludeStyle
  value: llvm
- key: modernize-replace-auto-ptr.IncludeStyle
  value: llvm
- key: modernize-use-nullptr.NullMacros
  value: 'NULL'
- key: readability-identifier-naming.NamespaceCase
  value: lower_case
- key: readability-identifier-naming.InlineNamespaceCase
  value: lower_case
- key: readability-identifier-naming.EnumConstantCase
  value: lower_case
- key: readability-identifier-naming.ConstantMemberCase
  value: lower_case
- key: readability-identifier-naming.PrivateMemberCase
  value: lower_case
- key: readability-identifier-naming.ProtectedMemberCase
  value: lower_case
- key: readability-identifier-naming.PublicMemberCase
  value: lower_case
- key: readability-identifier-naming.MemberCase
  value: lower_case
- key: readability-identifier-naming.ClassConstantCase
  value: lower_case
- key: readability-identifier-naming.ClassMemberCase
  value: lower_case
- key: readability-identifier-naming.GlobalConstantCase
  value: lower_case
- key: readability-identifier-naming.GlobalConstantPointerCase
  value: lower_case
- key: readability-identifier-naming.GlobalPointerCase
  value: lower_case
- key: readability-identifier-naming.GlobalVariableCase
  value: lower_case
- key: readability-identifier-naming.LocalConstantCase
  value: lower_case
- key: readability-identifier-naming.LocalConstantPointerCase
  value: lower_case
- key: readability-identifier-naming.LocalPointerCase
  value: lower_case
- key: readability-identifier-naming.LocalVariableCase
  value: lower_case
- key: readability-identifier-naming.StaticConstantCase
  value: lower_case
- key: readability-identifier-naming.StaticVariableCase
  value: lower_case
- key: readability-identifier-naming.ConstantCase
  value: lower_case
- key: readability-identifier-naming.VariableCase
  value: lower_case
- key: readability-identifier-naming.ConstantParameterCase
  value: lower_case
- key: readability-identifier-naming.ParameterPackCase
  value: lower_case
- key: readability-identifier-naming.ParameterCase
  value: lower_case
- key: readability-identifier-naming.PointerParameterCase
  value: lower_case
- key: readability-identifier-naming.ConstantPointerParameterCase
  value: lower_case
- key: readability-identifier-naming.AbstractClassCase
  value: lower_case
- key: readability-identifier-naming.StructCase
  value: lower_case
- key: readability-identifier-naming.ClassCase
  value: lower_case
- key: readability-identifier-naming.UnionCase
  value: lower_case
- key: readability-identifier-naming.EnumCase
  value: lower_case
- key: readability-identifier-naming.GlobalFunctionCase
  value: lower_case
- key: readability-identifier-naming.FunctionCase
  value: lower_case
- key: readability-identifier-naming.VirtualMethodCase
  value: lower_case
- key: readability-identifier-naming.PrivateMethodCase
  value: lower_case
- key: readability-identifier-naming.ProtectedMethodCase
  value: lower_case
- key: readability-identifier-naming.PublicMethodCase
  value: lower_case
- key: readability-identifier-naming.MethodCase
  value: lower_case
- key: readability-identifier-naming.TypedefCase
  value: lower_case
- key: readability-identifier-naming.ValueTemplateParameterCase
  value: CamelCase
- key: readability-identifier-naming.TemplateTemplateParameterCase
  value: CamelCase
- key: readability-identifier-naming.TypeAliasCase
  value: lower_case
- key: readability-identifier-naming.MacroDefinitionCase
  value: UPPER_CASE
- key: readability-identifier-naming.ObjcIvarCase
  value: lower_case
- key: readability-identifier-naming.NamespacePrefix
  value: ''
- key: readability-identifier-naming.InlineNamespacePrefix
  value: ''
- key: readability-identifier-naming.EnumConstantPrefix
  value: ''
- key: readability-identifier-naming.ConstexprVariablePrefix
  value: ''
- key: readability-identifier-naming.ConstantMemberPrefix
  value: ''
- key: readability-identifier-naming.PrivateMemberPrefix
  value: _
- key: readability-identifier-naming.ProtectedMemberPrefix
  value: _
- key: readability-identifier-naming.PublicMemberPrefix
  value: ''
- key: readability-identifier-naming.MemberPrefix
  value: ''
- key: readability-identifier-naming.ClassConstantPrefix
  value: ''
- key: readability-identifier-naming.ClassMemberPrefix
  value: ''
- key: readability-identifier-naming.GlobalConstantPrefix
  value: ''
- key: readability-identifier-naming.GlobalConstantPointerPrefix
  value: ''
- key: readability-identifier-naming.GlobalPointerPrefix
  value: ''
- key: readability-identifier-naming.GlobalVariablePrefix
  value: ''
- key: readability-identifier-naming.LocalConstantPrefix
  value: ''
- key: readability-identifier-naming.LocalConstantPointerPrefix
  value: ''
- key: readability-identifier-naming.LocalPointerPrefix
  value: ''
- key: readability-identifier-naming.LocalVariablePrefix
  value: ''
- key: readability-identifier-naming.StaticConstantPrefix
  value: ''
- key: readability-identifier-naming.StaticVariablePrefix
  value: ''
- key: readability-identifier-naming.ConstantPrefix
  value: ''
- key: readability-identifier-naming.VariablePrefix
  value: ''
- key: readability-identifier-naming.ConstantParameterPrefix
  value: ''
- key: readability-identifier-naming.ParameterPackPrefix
  value: ''
- key: readability-identifier-naming.ParameterPrefix
  value: ''
- key: readability-identifier-naming.PointerParameterPrefix
  value: ''
- key: readability-identifier-naming.ConstantPointerParameterPrefix
  value: ''
- key: readability-identifier-naming.AbstractClassPrefix
  value: ''
- key: readability-identifier-naming.StructPrefix
  value: ''
- key: readability-identifier-naming.ClassPrefix
  value: ''
- key: readability-identifier-naming.UnionPrefix
  value: ''
- key: readability-identifier-naming.EnumPrefix
  value: ''
- key: readability-identifier-naming.GlobalFunctionPrefix
  value: ''
- key: readability-identifier-naming.ConstexprFunctionPrefix
  value: ''
- key: readability-identifier-naming.FunctionPrefix
  value: ''
- key: readability-identifier-naming.ConstexprMethodPrefix
  value: ''
- key: readability-identifier-naming.VirtualMethodPrefix
  value: ''
- key: readability-identifier-naming.ClassMethodPrefix
  value: ''
- key: readability-identifier-naming.PrivateMethodPrefix
  value: ''
- key: readability-identifier-naming.ProtectedMethodPrefix
  value: ''
- key: readability-identifier-naming.PublicMethodPrefix
  value: ''
- key: readability-identifier-naming.MethodPrefix
  value: ''
- key: readability-identifier-naming.TypedefPrefix
  value: ''
- key: readability-identifier-naming.TypeTemplateParameterPrefix
  value: ''
- key: readability-identifier-naming.ValueTemplateParameterPrefix
  value: ''
- key: readability-identifier-naming.TemplateTemplateParameterPrefix
  value: ''
- key: readability-identifier-naming.TemplateParameterPrefix
  value: ''
- key: readability-identifier-naming.TypeAliasPrefix
  value: ''
- key: readability-identifier-naming.MacroDefinitionPrefix
  value: ''
- key: readability-identifier-naming.ObjcIvarPrefix
  value: ''
- key: readability-identifier-naming.NamespaceSuffix
  value: ''
- key: readability-identifier-naming.InlineNamespaceSuffix
  value: ''
- key: readability-identifier-naming.EnumConstantSuffix
  value: ''
- key: readability-identifier-naming.ConstexprVariableSuffix
  value: ''
- key: readability-identifier-naming.ConstantMemberSuffix
  value: ''
- key: readability-identifier-naming.PrivateMemberSuffix
  value: ''
- key: readability-identifier-naming.ProtectedMemberSuffix
  value: ''
- key: readability-identifier-naming.PublicMemberSuffix
  value: ''
- key: readability-identifier-naming.MemberSuffix
  value: ''
- key: readability-identifier-naming.ClassConstantSuffix
  value: ''
- key: readability-identifier-naming.ClassMemberSuffix
  value: ''
- key: readability-identifier-naming.GlobalConstantSuffix
  value: ''
- key: readability-identifier-naming.GlobalConstantPointerSuffix
  value: ''
- key: readability-identifier-naming.GlobalPointerSuffix
  value: ''
- key: readability-identifier-naming.GlobalVariableSuffix
  value: ''
- key: readability-identifier-naming.LocalConstantSuffix
  value: ''
- key: readability-identifier-naming.LocalConstantPointerSuffix
  value: ''
- key: readability-identifier-naming.LocalPointerSuffix
  value: ''
- key: readability-identifier-naming.LocalVariableSuffix
  value: ''
- key: readability-identifier-naming.StaticConstantSuffix
  value: ''
- key: readability-identifier-naming.StaticVariableSuffix
  value: ''
- key: readability-identifier-naming.ConstantSuffix
  value: ''
- key: readability-identifier-naming.VariableSuffix
  value: ''
- key: readability-identifier-naming.ConstantParameterSuffix
  value: ''
- key: readability-identifier-naming.ParameterPackSuffix
  value: ''
- key: readability-identifier-naming.ParameterSuffix
  value: ''
- key: readability-identifier-naming.PointerParameterSuffix
  value: ''
- key: readability-identifier-naming.ConstantPointerParameterSuffix
  value: ''
- key: readability-identifier-naming.AbstractClassSuffix
  value: ''
- key: readability-identifier-naming.StructSuffix
  value: ''
- key: readability-identifier-naming.ClassSuffix
  value: ''
- key: readability-identifier-naming.UnionSuffix
  value: ''
- key: readability-identifier-naming.EnumSuffix
  value: ''
- key: readability-identifier-naming.GlobalFunctionSuffix
  value: ''
- key: readability-identifier-naming.ConstexprFunctionSuffix
  value: ''
- key: readability-identifier-naming.FunctionSuffix
  value: ''
- key: readability-identifier-naming.ConstexprMethodSuffix
  value: ''
- key: readability-identifier-naming.VirtualMethodSuffix
  value: ''
- key: readability-identifier-naming.ClassMethodSuffix
  value: ''
- key: readability-identifier-naming.PrivateMethodSuffix
  value: ''
- key: readability-identifier-naming.ProtectedMethodSuffix
  value: ''
- key: readability-identifier-naming.PublicMethodSuffix
  value: ''
- key: readability-identifier-naming.MethodSuffix
  value: ''
- key: readability-identifier-naming.TypedefSuffix
  value: ''
- key: readability-identifier-naming.TypeTemplateParameterSuffix
  value: ''
- key: readability-identifier-naming.ValueTemplateParameterSuffix
  value: ''
- key: readability-identifier-naming.TemplateTemplateParameterSuffix
  value: ''
- key: readability-identifier-naming.TemplateParameterSuffix
  value: ''
- key: readability-identifier-naming.TypeAliasSuffix
  value: ''
- key: readability-identifier-naming.MacroDefinitionSuffix
  value: ''
- key: readability-identifier-naming.ObjcIvarSuffix
  value: ''
- key: readability-identifier-naming.FunctionIgnoredRegexp
  value: LLVMFuzzerTestOneInput
```

**Comment by wpoch** [comment]
Same error over here with the Mac VSCode:

![image](https://user-images.githubusercontent.com/308915/165394090-c68ff034-e9d2-4661-a866-a930d7849a54.png)

```
uname -a

Darwin xxxxx 21.4.0 Darwin Kernel Version 21.4.0: Fri Mar 18 00:45:05 PDT 2022; root:xnu-8020.101.4~15/RELEASE_X86_64 x86_64

```


```
Version: 1.66.2
Commit: dfd34e8260c270da74b5c2d86d61aee4b6d56977
Date: 2022-04-11T07:49:20.994Z
Electron: 17.2.0
Chromium: 98.0.4758.109
Node.js: 16.13.0
V8: 9.8.177.11-electron.0
OS: Darwin x64 21.4.0
```

Sadly I cannot share the Yaml file due to confidential reasons ¯\_(ツ)_/¯

**Comment by pascalre** [comment]
@johnmcfarlane @wpoch can you please provide some more information on this issue? I can not reproduce it (working on a M1 Mac).

```
uname -a

Darwin xxxxx 21.4.0 Darwin Kernel Version 21.4.0: Fri Mar 18 00:47:26 PDT 2022; root:xnu-8020.101.4~15/RELEASE_ARM64_T8101 arm64
```

I need more information about the folder where you are executing recursive sort:
- Does it contain subfolders?
- Does it contain multiple yaml files?
- What are the extensions of the yaml files?
- Please provide a sample folder which causes the error.

And btw: The yaml you posted will never be sorted, since it provides only a yaml list. List will not sorted with this extension.

**Comment by johnmcfarlane** [comment]
* Does it contain subfolders?
Yes

* Does it contain multiple yaml files?
Yes, but not with a .yml/.yaml extension

* What are the extensions of the yaml files?
Filename is .clang-tidy

* Please provide a sample folder which causes the error.
.clang-tidy in [this folder](https://github.com/johnmcfarlane/wss/) is what I was trying to apply this to.

**Comment by pascalre** [comment]
@johnmcfarlane @wpoch I just released version 6.0.2 where I updated the algorithm for recursive sorting. Can you check, if the issue is still present?

**Comment by johnmcfarlane** [comment]
Hard to tell as there is a different message:
![image](https://user-images.githubusercontent.com/734007/211909866-85e752d5-e36c-4f49-8b2e-4a1362e82a05.png)
